src/librbd/internal.cc

```diff
@@ -2872,6 +2872,55 @@ int validate_mirroring_enabled(ImageCtx *ictx) {
             << cpp_strerror(r) << dendl;
         }
       }
+    } else if (next_mirror_mode == cls::rbd::MIRROR_MODE_DISABLED) {
+
+      bool needs_rollback = false;
+      if (current_mirror_mode == cls::rbd::MIRROR_MODE_POOL) {
+        r = cls_client::mirror_mode_set(&io_ctx, cls::rbd::MIRROR_MODE_IMAGE);
+        if (r < 0) {
+          lderr(cct) << "Failed to set mirror mode to image: "
+            << cpp_strerror(r) << dendl;
+          return r;
+        }
+        needs_rollback = true;
+      }
+
+      std::vector<std::string> image_ids;
+      r = cls_client::mirror_image_list(&io_ctx, &image_ids);
```

Comment:
There is the pool setting and the per-image setting (and really the per-image setting isn't a setting, it's the advertisement in the rbd_mirroring directory).

If you have explicit control over which images are mirrored, you set the pool to image-mode.  If you want all images to automatically be mirrored (assuming the images have journaling enabled), set it to image-mode.  When in pool-mode, it really doesn't make sense to allow the user to run "rbd mirror image enable / disable" so we should prohibit it.  Similarly, if you are disabling image mode but you haven't already manually disabled mirroring on all images, we should remind them to do it.